Makefile

```diff
@@ -2,7 +2,8 @@ develop:
 	python setup.py develop
 
 lint flake: checkrst  pyroma bandit mypy
-	flake8 janus tests
+	# F401 is upset about typing.Deque, List, and Set which are used in hinting
```

Comment:
The type annotations appear to be ancient. Could we just update them to modern annotations and not disable this?

i.e. `self._queue: Deque[T] = deque()`

Reply:
Addressed!

Reply:
I also upgraded from `typing.List` to the builtin `list`, as https://peps.python.org/pep-0585/ is supported in 3.9+